tux/cogs/moderation/purge.py

```diff
@@ -0,0 +1,79 @@
+import discord
+from discord import app_commands
+from discord.ext import commands
+from loguru import logger
+
+
+class Purge(commands.Cog):
+    def __init__(self, bot: commands.Bot) -> None:
+        self.bot = bot
+
+    async def send_embed(
+        self,
+        interaction: discord.Interaction,
+        title: str,
+        description: str,
+        color: discord.Colour,
+        error_info: str | None = None,
+    ) -> None:
+        embed = discord.Embed(
+            title=title, description=description, color=color, timestamp=interaction.created_at
+        )
+        if error_info:
+            embed.add_field(name="Error Details", value=f"`{error_info}`", inline=False)
+        embed.set_footer(
+            text=f"Requested by {interaction.user.display_name}",
+            icon_url=interaction.user.display_avatar.url,
+        )
+        await interaction.followup.send(embed=embed)
+
+    @app_commands.checks.has_any_role("Admin", "Sr. Mod", "Mod", "Jr. Mod")
+    @app_commands.command(
+        name="purge", description="Deletes a set number of messages in a channel."
+    )
+    @app_commands.describe(number_messages="The number of messages to be purged.")
+    async def purge_messages(
+        self, interaction: discord.Interaction, number_messages: int = 10
+    ) -> None:
+        if not interaction.channel or interaction.channel.type != discord.ChannelType.text:
```

Comment:
This is unnecessary, sometimes you want to purge messages in a VC's text chat actually